app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,14 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             grep_str = params[:grep].to_s
+             context_int = if params[:context].respond_to?(:to_i)
```

Comment:
Also very minor, but this might look better if written as a guard instead, so

```ruby
unless params[:context].respond_to?(:to_i)
    # flash, redirect, return
end
```

and then in the interpolation use `params[:context].to_i`.